### PR TITLE
Analytics hub tracks

### LINKED ISSUE #7588: [Analytics Hub] Add Analytics Events
Please see pe5pgL-153-p2

- [x] Update `used_analytics`
- [x] Add the other new events (see post)

### PR DESCRIPTION
**Pull request by ThomazFB**
Summary
==========
Fix issue #7588 by introducing a set of tracks to the Analytics Hub group of features.

Expect the following tracks to be introduced here:

`_dashboard_see_more_analytics_tapped`: Tapping on “See more” in the My Store screen.
`_analytics_hub_date_range_button_tapped`: Tapping on the date range selector button.
`_analytics_hub_date_range_option_selected`: 
    - Custom properties: `option`: The option selected like “today”, “yesterday”, or “custom”. Make sure that this is not translated into the local language.
`_analytics_hub_see_report_tapped`: Tapping the “See report” button.
    - Custom properties:`card`: The card type in English, e.g., “revenue”, “orders”.

How to Test
==========
1. Open the Woo app with the Analytics Hub feature flag enabled
2. Click on the `See more` button inside the My Store stats
3. Verify that the Analytics Hub opens, click the date selector on the top right corner and change it
4. Click on the `See Report` button
5. Return from the browser back to the Analytics Hub

After doing all this flow, verify inside the logs the presence of the tracks as described in the Summary.

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/analytics/AnalyticsViewModel.kt [hunk 1]
```diff
@@ -443,6 +454,24 @@ class AnalyticsViewModel @Inject constructor(
...
+    private fun trackSeeReportClicked(selectedCardType: String) {
+        AnalyticsTracker.track(
+            AnalyticsEvent.ANALYTICS_HUB_SEE_REPORT_TAPPED,
+            mapOf(
+                AnalyticsTracker.KEY_CARD to selectedCardType
+            )
+        )
+    }
+
+    private fun trackSelectedDateRange(selectedTimePeriod: AnalyticTimePeriod) {
```
Comment (nbradbury) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/analytics/AnalyticsViewModel.kt:468]: This isn't something that should hold up this PR, but I'm wondering if we should track what custom date ranges people are selecting?

Comment (nbradbury) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/analytics/AnalyticsViewModel.kt:468]: **Update:** While testing the app, I noticed that nothing is tracked when the user chooses a custom date range.

Comment (ThomazFB) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/analytics/AnalyticsViewModel.kt:468]: Oh, thanks for catching that, @nbradbury! I'll submit a fix soon, but before that, I would like @shiki inputs on your suggestion to log the selected range in custom mode. I think it makes perfect sense to me.

Comment (shiki) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/analytics/AnalyticsViewModel.kt:468]: I asked this myself before, but I didn't know how we're gonna use the data. So I took it out of the scope. 🤪 I didn't wanna add too many tasks for you. I guess we can put it off for later if we realize we'd really need it.

Comment (ThomazFB) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/analytics/AnalyticsViewModel.kt:468]: Thanks for sharing your thoughts, @shiki! In that case, I submitted just the fix for the custom selection not getting tracked @nbradbury right here: https://github.com/woocommerce/woocommerce-android/pull/7649/commits/7b83e4e8cde3680c31db7789c1931476af0e94e8

Let me know what you think!

**Comment by ThomazFB** [comment]
@nbradbury, thanks for reviewing the tracks! The PR is now ready for a second round!

**Review by nbradbury — APPROVED** [review]
:shipit:

**Pull request merged**